mlflow/openai/__init__.py

```diff
@@ -0,0 +1,483 @@
+    :param task: The task the model is performing, e.g., ``openai.ChatCompletion`` or
+                 ``'chat.completions'``.
+    :param conda_env: {{ conda_env }}
+    :param code_paths: A list of local filesystem paths to Python file dependencies (or directories
+                       containing file dependencies). These files are *prepended* to the system
+                       path when the model is loaded.
+    :param mlflow_model: :py:mod:`mlflow.models.Model` this flavor is being added to.
+
+    :param signature: :py:class:`ModelSignature <mlflow.models.ModelSignature>`
+                      describes model input and output :py:class:`Schema <mlflow.types.Schema>`.
+                      The model signature can be :py:func:`inferred <mlflow.models.infer_signature>`
+                      from datasets with valid model input (e.g. the training dataset with target
+                      column omitted) and valid model output (e.g. model predictions generated on
+                      the training dataset), for example:
+
+                      .. code-block:: python
+
+                        from mlflow.models.signature import infer_signature
+
+                        train = df.drop_column("target_label")
+                        predictions = ...  # compute model predictions
+                        signature = infer_signature(train, predictions)
+    :param input_example: Input example provides one or several instances of valid
+                          model input. The example can be used as a hint of what data to feed the
+                          model. The given example will be converted to a Pandas DataFrame and then
+                          serialized to json using the Pandas split-oriented format. Bytes are
+                          base64-encoded.
+    :param pip_requirements: {{ pip_requirements }}
+    :param extra_pip_requirements: {{ extra_pip_requirements }}
+    :param metadata: Custom metadata dictionary passed to the model and stored in the MLmodel file.
+
+                     .. Note:: Experimental: This parameter may change or be removed in a future
+                                             release without warning.
+    :param  kwargs: Keyword arguments specific to the OpenAI task, such as the ``temperature`` or
+                    or ``top_p`` value to use for chat completion.
+    """
+    _validate_env_arguments(conda_env, pip_requirements, extra_pip_requirements)
+    path = os.path.abspath(path)
+    _validate_and_prepare_target_save_path(path)
+    code_dir_subpath = _validate_and_copy_code_paths(code_paths, path)
+    task = _get_task_name(task)
+
+    if mlflow_model is None:
+        mlflow_model = Model()
+    if signature is not None:
+        mlflow_model.signature = signature
+    else:
+        if task == "chat.completions":
+            mlflow_model.signature = ModelSignature(
+                inputs=Schema([ColSpec(type="string", name="messages")]),
```

Comment:
Does `ColSpec` support a list of dictionaries?

Reply:
```suggestion
                inputs=Schema([ColSpec(type=List[Dict[str, str]], name="messages")]),
```

once https://github.com/mlflow/mlflow/pull/8181 is merged